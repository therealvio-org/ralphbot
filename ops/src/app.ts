import { App as BaseApp, AppProps as BaseAppProps } from "aws-cdk-lib"

export interface AppExtensionProps extends BaseAppProps {
  projectName: string
  commit: string
  environment: string
  version: string
}

export class AppExtension extends BaseApp {
  public readonly commit: string
  public readonly environment: string
  public readonly projectName: string
  public readonly version: string

  constructor(props: AppExtensionProps) {
    super(props)

    this.projectName = props.projectName
    this.commit = props.commit
    this.environment = props.environment
    this.version = props.version
  }
}
