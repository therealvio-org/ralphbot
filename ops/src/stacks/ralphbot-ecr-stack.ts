import { Construct } from "constructs"
import { Stack, StackProps } from "aws-cdk-lib"
import { aws_ecr as ecr } from "aws-cdk-lib"

export class RalphbotEcrStack extends Stack {
  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props)

    new ecr.Repository(this, "ralphbotECRRepository", {
      lifecycleRules: [
        {
          maxImageCount: 5,
        },
      ],
      repositoryName: "ralphbot/master",
    })
  }
}
