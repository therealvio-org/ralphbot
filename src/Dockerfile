FROM public.ecr.aws/docker/library/golang:1.18.3 as build_image

WORKDIR /app

COPY . ./

RUN go mod download && go mod verify
RUN CGO_ENABLED=0 go build -v -o /tmp/build/ralphbot/ ./...
#CGO_ENABLED=0 to force a static build. Next stage of build uses an image that has libraries in different locations
#https://stackoverflow.com/a/36308464

FROM public.ecr.aws/docker/library/golang:1.18.3-alpine
WORKDIR /app
COPY --from=build_image /tmp/build/ralphbot/ ./
COPY --from=build_image /app/internal/dadjoke/jokes.json ./
CMD [ "./ralphbot" ]