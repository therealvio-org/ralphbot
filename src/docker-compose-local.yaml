version: "3.8"
services:
  ralphbot:
    build:
      context: "${PWD}"
      dockerfile: Dockerfile
    container_name: ralphbot
    environment:
      - BOT_TOKEN
      - GUILD_ID
      - COMMIT
      - VERSION
      - ENVIRONMENT
