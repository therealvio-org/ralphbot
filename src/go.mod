module ralphbot

go 1.22.1

require (
	github.com/bwmarrin/discordgo v0.27.1
	github.com/kelseyhightower/envconfig v1.4.0
	github.com/stretchr/testify v1.9.0
)

require (
	github.com/davecgh/go-spew v1.1.1 // indirect
	github.com/gorilla/websocket v1.5.0 // indirect
	github.com/pmezard/go-difflib v1.0.0 // indirect
	golang.org/x/crypto v0.17.0 // indirect
	golang.org/x/sys v0.15.0 // indirect
	gopkg.in/yaml.v3 v3.0.1 // indirect
)
