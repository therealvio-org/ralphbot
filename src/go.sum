github.com/bwmarrin/discordgo v0.27.1 h1:ib9AIc/dom1E/fSIulrBwnez0CToJE113ZGt4HoliGY=
github.com/bwmarrin/discordgo v0.27.1/go.mod h1:NJZpH+1AfhIcyQsPeuBKsUtYrRnjkyu0kIVMCHkZtRY=
github.com/davecgh/go-spew v1.1.1 h1:vj9j/u1bqnvCEfJOwUhtlOARqs3+rkHYY13jYWTU97c=
github.com/davecgh/go-spew v1.1.1/go.mod h1:J7Y8YcW2NihsgmVo/mv3lAwl/skON4iLHjSsI+c5H38=
github.com/gorilla/websocket v1.4.2/go.mod h1:YR8l580nyteQvAITg2hZ9XVh4b55+EU/adAjf1fMHhE=
github.com/gorilla/websocket v1.5.0 h1:PPwGk2jz7EePpoHN/+ClbZu8SPxiqlu12wZP/3sWmnc=
github.com/gorilla/websocket v1.5.0/go.mod h1:YR8l580nyteQvAITg2hZ9XVh4b55+EU/adAjf1fMHhE=
github.com/kelseyhightower/envconfig v1.4.0 h1:Im6hONhd3pLkfDFsbRgu68RDNkGF1r3dvMUtDTo2cv8=
github.com/kelseyhightower/envconfig v1.4.0/go.mod h1:cccZRl6mQpaq41TPp5QxidR+Sa3axMbJDNb//FQX6Gg=
github.com/pmezard/go-difflib v1.0.0 h1:4DBwDE0NGyQoBHbLQYPwSUPoCMWR5BEzIk/f1lZbAQM=
github.com/pmezard/go-difflib v1.0.0/go.mod h1:iKH77koFhYxTK1pcRnkKkqfTogsbg7gZNVY4sRDYZ/4=
github.com/stretchr/testify v1.9.0 h1:HtqpIVDClZ4nwg75+f6Lvsy/wHu+3BoSGCbBAcpTsTg=
github.com/stretchr/testify v1.9.0/go.mod h1:r2ic/lqez/lEtzL7wO/rwa5dbSLXVDPFyf8C91i36aY=
golang.org/x/crypto v0.0.0-20210421170649-83a5a9bb288b/go.mod h1:T9bdIzuCu7OtxOm1hfPfRQxPLYneinmdGuTeoZ9dtd4=
golang.org/x/crypto v0.17.0 h1:r8bRNjWL3GshPW3gkd+RpvzWrZAwPS49OmTGZ/uhM4k=
golang.org/x/crypto v0.17.0/go.mod h1:gCAAfMLgwOJRpTjQ2zCCt2OcSfYMTeZVSRtQlPC7Nq4=
golang.org/x/net v0.0.0-20210226172049-e18ecbb05110/go.mod h1:m0MpNAwzfU5UDzcl9v0D8zg8gWTRqZa9RBIspLL5mdg=
golang.org/x/sys v0.0.0-20201119102817-f84b799fce68/go.mod h1:h1NjWce9XRLGQEsW7wpKNCjG9DtNlClVuFLEZdDNbEs=
golang.org/x/sys v0.15.0 h1:h48lPFYpsTvQJZF4EKyI4aLHaev3CxivZmv7yZig9pc=
golang.org/x/sys v0.15.0/go.mod h1:/VUhepiaJMQUp4+oa/7Zr1D23ma6VTLIYjOOTFZPUcA=
golang.org/x/term v0.0.0-20201126162022-7de9c90e9dd1/go.mod h1:bj7SfCRtBDWHUb9snDiAeCFNEtKQo2Wmx5Cou7ajbmo=
golang.org/x/text v0.3.3/go.mod h1:5Zoc/QRtKVWzQhOtBMvqHzDpF6irO9z98xDceosuGiQ=
golang.org/x/tools v0.0.0-20180917221912-90fa682c2a6e/go.mod h1:n7NCudcB/nEzxVGmLbDWY5pfWTLqBcC2KZ6jyYvM4mQ=
gopkg.in/check.v1 v0.0.0-20161208181325-20d25e280405 h1:yhCVgyC4o1eVCa2tZl7eS0r+SDo693bJlVdllGtEeKM=
gopkg.in/check.v1 v0.0.0-20161208181325-20d25e280405/go.mod h1:Co6ibVJAznAaIkqp8huTwlJQCZ016jof/cbN4VW5Yz0=
gopkg.in/yaml.v3 v3.0.1 h1:fxVm/GzAzEWqLHuvctI91KS9hhNmmWOoWu0XTYJS7CA=
gopkg.in/yaml.v3 v3.0.1/go.mod h1:K4uyk7z7BCEPqu6E+C64Yfv1cQ7kz7rIZviUmN+EgEM=
