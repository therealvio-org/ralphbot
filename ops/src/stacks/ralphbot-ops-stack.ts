import * as ec2 from "aws-cdk-lib/aws-ec2"
import * as ecr from "aws-cdk-lib/aws-ecr"
import * as ecs from "aws-cdk-lib/aws-ecs"
import * as iam from "aws-cdk-lib/aws-iam"
import * as logs from "aws-cdk-lib/aws-logs"
import * as secretsmanager from "aws-cdk-lib/aws-secretsmanager"
import * as ssm from "aws-cdk-lib/aws-ssm"
import { AppExtensionProps } from "../app"
import { Construct } from "constructs"
import { Stack, StackProps } from "aws-cdk-lib/core"

export class RalphbotStack extends Stack {
  constructor(
    scope: Construct,
    id: string,
    props?: StackProps,
    extendedProps?: AppExtensionProps
  ) {
    super(scope, id, props)

    const vpc = new ec2.Vpc(this, "vpc", {
      maxAzs: 1,
      natGateways: 0,
    })

    const repositoryRef = ecr.Repository.fromRepositoryArn(
      this,
      "ralphbotECRRepositoryRef",
      `arn:aws:ecr:${props?.env?.region}:${props?.env?.account}:repository/ralphbot/master`
    )

    const ralphbotImage = ecs.ContainerImage.fromEcrRepository(
      repositoryRef,
      extendedProps?.version
    )

    const cluster = new ecs.Cluster(this, "Cluster", {
      vpc: vpc,
    })
    const taskDefinition = new ecs.FargateTaskDefinition(
      this,
      "TaskDefinition",
      {
        cpu: 256,
        memoryLimitMiB: 512,
        runtimePlatform: {
          cpuArchitecture: ecs.CpuArchitecture.ARM64,
        },
      }
    )

    const botToken = secretsmanager.Secret.fromSecretNameV2(
      this,
      "botTokenFromName",
      "ralphbot/token"
    )
    const secretArnSuffix = ssm.StringParameter.valueForStringParameter(
      this,
      "/ralphbot/token/arn-suffix"
    )

    taskDefinition.addToExecutionRolePolicy(
      new iam.PolicyStatement({
        actions: [
          "secretsmanager:GetSecretValue",
          "secretsmanager:DescribeSecret",
        ],
        resources: [`${botToken.secretArn}-${secretArnSuffix}`],
      })
    )

    taskDefinition.addContainer("ralphbotContainer", {
      image: ralphbotImage,
      environment: {
        // trigger removal of registered commands on bot startup
        // this is configured to true on the app code itself, we're surfacing it here so it remains unobscured
        REMOVE_COMMANDS: "true",
      },
      secrets: {
        BOT_TOKEN: ecs.Secret.fromSecretsManager(botToken, "token"),
      },
      logging: ecs.LogDriver.awsLogs({
        streamPrefix: "ralphbot",
        logRetention: logs.RetentionDays.ONE_WEEK,
      }),
    })
    new ecs.FargateService(this, "Service", {
      assignPublicIp: true,
      cluster: cluster,
      taskDefinition: taskDefinition,
      circuitBreaker: { rollback: true },
      vpcSubnets: vpc.selectSubnets({
        subnetType: ec2.SubnetType.PUBLIC,
      }),
    })
  }
}
