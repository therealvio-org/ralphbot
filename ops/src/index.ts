#!/usr/bin/env node
import "source-map-support/register"
import * as cdk from "aws-cdk-lib"
import { AppExtension } from "./app"
import { RalphbotEcrStack } from "./stacks/ralphbot-ecr-stack"
import { RalphbotStack } from "./stacks/ralphbot-ops-stack"

const {
  COMMIT: commit,
  ENVIRONMENT: environment,
  VERSION: version,
  AWS_ACCOUNT_ID: accountId,
  REGION: region,
} = process.env

const projectName = `ralphbot`

if (!environment) {
  throw new Error("ENVIRONMENT environment variable must be supplied")
}

if (!commit) {
  throw new Error("COMMIT environment variable must be supplied")
}

if (!version) {
  throw new Error("VERSION environment variable must be supplied")
}

const extendedProperties = new AppExtension({
  commit,
  environment,
  projectName,
  version,
})

const app = new cdk.App()
new RalphbotEcrStack(app, `${projectName}-ecr`)
new RalphbotStack(
  app,
  `${projectName}-ops`,
  {
    env: {
      account: accountId,
      region: region,
    },
  },
  extendedProperties
)
