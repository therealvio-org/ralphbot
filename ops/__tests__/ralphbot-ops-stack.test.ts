import * as cdk from "aws-cdk-lib"
import { AppExtension } from "../src/app"
import { RalphbotStack } from "../src/stacks/ralphbot-ops-stack"
import { Template } from "aws-cdk-lib/assertions"

describe("Retrieve ralphbot ECR Stack", () => {
  const app = new cdk.App()
  const projectName = `ralphbot`

  const commit = "aaa123"
  const environment = "test"
  const version = "2.0.20"
  const accountId = "12345678901234"
  const region = "foo-bar-1"

  const extendedProperties = new AppExtension({
    commit,
    environment,
    projectName,
    version,
  })

  const stack = new RalphbotStack(
    app,
    `${projectName}-ops`,
    {
      env: {
        account: accountId,
        region: region,
      },
    },
    extendedProperties,
  )

  const template = Template.fromStack(stack)

  it("Synths and snapshots", () => {
    expect(template).toMatchSnapshot()
  })
})
