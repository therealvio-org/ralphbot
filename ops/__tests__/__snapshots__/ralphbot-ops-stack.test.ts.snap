// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Retrieve ralphbot ECR Stack Synths and snapshots 1`] = `
Object {
  "Parameters": Object {
    "BootstrapVersion": Object {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
    "SsmParameterValueralphbottokenarnsuffixC96584B6F00A464EAD1953AFF4B05118Parameter": Object {
      "Default": "/ralphbot/token/arn-suffix",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": Object {
    "ClusterEB0386A7": Object {
      "Type": "AWS::ECS::Cluster",
    },
    "ServiceD69D759B": Object {
      "Properties": Object {
        "Cluster": Object {
          "Ref": "ClusterEB0386A7",
        },
        "DeploymentConfiguration": Object {
          "DeploymentCircuitBreaker": Object {
            "Enable": true,
            "Rollback": true,
          },
          "MaximumPercent": 200,
          "MinimumHealthyPercent": 50,
        },
        "DeploymentController": Object {
          "Type": "ECS",
        },
        "EnableECSManagedTags": false,
        "LaunchType": "FARGATE",
        "NetworkConfiguration": Object {
          "AwsvpcConfiguration": Object {
            "AssignPublicIp": "ENABLED",
            "SecurityGroups": Array [
              Object {
                "Fn::GetAtt": Array [
                  "ServiceSecurityGroupC96ED6A7",
                  "GroupId",
                ],
              },
            ],
            "Subnets": Array [
              Object {
                "Ref": "vpcPublicSubnet1Subnet2E65531E",
              },
            ],
          },
        },
        "TaskDefinition": Object {
          "Ref": "TaskDefinitionB36D86D9",
        },
      },
      "Type": "AWS::ECS::Service",
    },
    "ServiceSecurityGroupC96ED6A7": Object {
      "Properties": Object {
        "GroupDescription": "ralphbot-ops/Service/SecurityGroup",
        "SecurityGroupEgress": Array [
          Object {
            "CidrIp": "0.0.0.0/0",
            "Description": "Allow all outbound traffic by default",
            "IpProtocol": "-1",
          },
        ],
        "VpcId": Object {
          "Ref": "vpcA2121C38",
        },
      },
      "Type": "AWS::EC2::SecurityGroup",
    },
    "TaskDefinitionB36D86D9": Object {
      "Properties": Object {
        "ContainerDefinitions": Array [
          Object {
            "Essential": true,
            "Image": Object {
              "Fn::Join": Array [
                "",
                Array [
                  "12345678901234.dkr.ecr.foo-bar-1.",
                  Object {
                    "Ref": "AWS::URLSuffix",
                  },
                  "/ralphbot/master:2.0.20",
                ],
              ],
            },
            "LogConfiguration": Object {
              "LogDriver": "awslogs",
              "Options": Object {
                "awslogs-group": Object {
                  "Ref": "TaskDefinitionralphbotContainerLogGroup2050EE52",
                },
                "awslogs-region": "foo-bar-1",
                "awslogs-stream-prefix": "ralphbot",
              },
            },
            "Name": "ralphbotContainer",
            "Secrets": Array [
              Object {
                "Name": "BOT_TOKEN",
                "ValueFrom": Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":secretsmanager:foo-bar-1:12345678901234:secret:ralphbot/token:token::",
                    ],
                  ],
                },
              },
            ],
          },
        ],
        "Cpu": "256",
        "ExecutionRoleArn": Object {
          "Fn::GetAtt": Array [
            "TaskDefinitionExecutionRole8D61C2FB",
            "Arn",
          ],
        },
        "Family": "ralphbotopsTaskDefinition7E1521A2",
        "Memory": "512",
        "NetworkMode": "awsvpc",
        "RequiresCompatibilities": Array [
          "FARGATE",
        ],
        "TaskRoleArn": Object {
          "Fn::GetAtt": Array [
            "TaskDefinitionTaskRoleFD40A61D",
            "Arn",
          ],
        },
      },
      "Type": "AWS::ECS::TaskDefinition",
    },
    "TaskDefinitionExecutionRole8D61C2FB": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "ecs-tasks.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "TaskDefinitionExecutionRoleDefaultPolicy1F3406F5": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "secretsmanager:GetSecretValue",
                "secretsmanager:DescribeSecret",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "arn:",
                    Object {
                      "Ref": "AWS::Partition",
                    },
                    ":secretsmanager:foo-bar-1:12345678901234:secret:ralphbot/token-",
                    Object {
                      "Ref": "SsmParameterValueralphbottokenarnsuffixC96584B6F00A464EAD1953AFF4B05118Parameter",
                    },
                  ],
                ],
              },
            },
            Object {
              "Action": Array [
                "ecr:BatchCheckLayerAvailability",
                "ecr:GetDownloadUrlForLayer",
                "ecr:BatchGetImage",
              ],
              "Effect": "Allow",
              "Resource": "arn:aws:ecr:foo-bar-1:12345678901234:repository/ralphbot/master",
            },
            Object {
              "Action": "ecr:GetAuthorizationToken",
              "Effect": "Allow",
              "Resource": "*",
            },
            Object {
              "Action": Array [
                "logs:CreateLogStream",
                "logs:PutLogEvents",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "TaskDefinitionralphbotContainerLogGroup2050EE52",
                  "Arn",
                ],
              },
            },
            Object {
              "Action": Array [
                "secretsmanager:GetSecretValue",
                "secretsmanager:DescribeSecret",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "arn:",
                    Object {
                      "Ref": "AWS::Partition",
                    },
                    ":secretsmanager:foo-bar-1:12345678901234:secret:ralphbot/token-??????",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "TaskDefinitionExecutionRoleDefaultPolicy1F3406F5",
        "Roles": Array [
          Object {
            "Ref": "TaskDefinitionExecutionRole8D61C2FB",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "TaskDefinitionTaskRoleFD40A61D": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "ecs-tasks.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "TaskDefinitionralphbotContainerLogGroup2050EE52": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "RetentionInDays": 7,
      },
      "Type": "AWS::Logs::LogGroup",
      "UpdateReplacePolicy": "Retain",
    },
    "vpcA2121C38": Object {
      "Properties": Object {
        "CidrBlock": "10.0.0.0/16",
        "EnableDnsHostnames": true,
        "EnableDnsSupport": true,
        "InstanceTenancy": "default",
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "ralphbot-ops/vpc",
          },
        ],
      },
      "Type": "AWS::EC2::VPC",
    },
    "vpcIGWE57CBDCA": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "ralphbot-ops/vpc",
          },
        ],
      },
      "Type": "AWS::EC2::InternetGateway",
    },
    "vpcIsolatedSubnet1RouteTable0D6B2D3D": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "ralphbot-ops/vpc/IsolatedSubnet1",
          },
        ],
        "VpcId": Object {
          "Ref": "vpcA2121C38",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "vpcIsolatedSubnet1RouteTableAssociation172210D4": Object {
      "Properties": Object {
        "RouteTableId": Object {
          "Ref": "vpcIsolatedSubnet1RouteTable0D6B2D3D",
        },
        "SubnetId": Object {
          "Ref": "vpcIsolatedSubnet1Subnet8B28CEB3",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "vpcIsolatedSubnet1Subnet8B28CEB3": Object {
      "Properties": Object {
        "AvailabilityZone": "dummy1a",
        "CidrBlock": "10.0.128.0/17",
        "MapPublicIpOnLaunch": false,
        "Tags": Array [
          Object {
            "Key": "aws-cdk:subnet-name",
            "Value": "Isolated",
          },
          Object {
            "Key": "aws-cdk:subnet-type",
            "Value": "Isolated",
          },
          Object {
            "Key": "Name",
            "Value": "ralphbot-ops/vpc/IsolatedSubnet1",
          },
        ],
        "VpcId": Object {
          "Ref": "vpcA2121C38",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "vpcPublicSubnet1DefaultRoute10708846": Object {
      "DependsOn": Array [
        "vpcVPCGW7984C166",
      ],
      "Properties": Object {
        "DestinationCidrBlock": "0.0.0.0/0",
        "GatewayId": Object {
          "Ref": "vpcIGWE57CBDCA",
        },
        "RouteTableId": Object {
          "Ref": "vpcPublicSubnet1RouteTable48A2DF9B",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "vpcPublicSubnet1RouteTable48A2DF9B": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "ralphbot-ops/vpc/PublicSubnet1",
          },
        ],
        "VpcId": Object {
          "Ref": "vpcA2121C38",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "vpcPublicSubnet1RouteTableAssociation5D3F4579": Object {
      "Properties": Object {
        "RouteTableId": Object {
          "Ref": "vpcPublicSubnet1RouteTable48A2DF9B",
        },
        "SubnetId": Object {
          "Ref": "vpcPublicSubnet1Subnet2E65531E",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "vpcPublicSubnet1Subnet2E65531E": Object {
      "Properties": Object {
        "AvailabilityZone": "dummy1a",
        "CidrBlock": "10.0.0.0/17",
        "MapPublicIpOnLaunch": true,
        "Tags": Array [
          Object {
            "Key": "aws-cdk:subnet-name",
            "Value": "Public",
          },
          Object {
            "Key": "aws-cdk:subnet-type",
            "Value": "Public",
          },
          Object {
            "Key": "Name",
            "Value": "ralphbot-ops/vpc/PublicSubnet1",
          },
        ],
        "VpcId": Object {
          "Ref": "vpcA2121C38",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "vpcVPCGW7984C166": Object {
      "Properties": Object {
        "InternetGatewayId": Object {
          "Ref": "vpcIGWE57CBDCA",
        },
        "VpcId": Object {
          "Ref": "vpcA2121C38",
        },
      },
      "Type": "AWS::EC2::VPCGatewayAttachment",
    },
  },
  "Rules": Object {
    "CheckBootstrapVersion": Object {
      "Assertions": Array [
        Object {
          "Assert": Object {
            "Fn::Not": Array [
              Object {
                "Fn::Contains": Array [
                  Array [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  Object {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
